pyfr/solvers/base/system.py

```diff
@@ -152,16 +154,29 @@ def _load_bc_inters(self, mesh, elemap):
         bccls = self.bbcinterscls
         bcmap = {b.type: b for b in subclasses(bccls, just_leaf=True)}
 
+        # Iterate over all BCs (including ones not on this rank)
         bc_inters = []
-        for bname, interarr in mesh.bcon.items():
+        bcnames = [s[9:] for s in self.cfg.sections() if s[:9] == 'soln-bcs-']
```

Comment:
Don't rely on the config file for the master list of BCs across all ranks.  Use the mesh codec.